repos/system_upgrade/el7toel8/actors/checkkdegnome/libraries/library.py

```diff
@@ -0,0 +1,66 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledDesktopsFacts, InstalledKdeAppsFacts
+
+
+def check_kde_gnome():
+    desktopFacts = next(api.consume(InstalledDesktopsFacts))
+    kde_desktop_installed = desktopFacts.kde_installed
+    gnome_desktop_installed = desktopFacts.gnome_installed
+
+    # No desktop installed, we don't even care about apps as they are most likely not used or even installed
+    if not kde_desktop_installed and not gnome_desktop_installed:
+        api.current_logger().info("No desktop installed. Continuing with the upgrade.")
+        return
+
+    if kde_desktop_installed:
+        api.current_logger().info("KDE desktop is installed. Checking what we can do about it.")
+        if not gnome_desktop_installed:
+            api.current_logger().error("Cannot perform the upgrade because there is"
+                                       " no other desktop than KDE installed.")
+            # We cannot continue with the upgrade process
+            reporting.create_report([
+                reporting.Title("Cannot upgrade because there is no other desktop than KDE installed."),
+                reporting.Summary("With only KDE installed, there would be no other desktop env. after upgrade."),
```

Comment:
```suggestion
                reporting.Summary("The KDE desktop environment is not available on RHEL 8. The KDE-related packages will be uninstalled during the upgrade and because the only currently installed desktop environment is KDE, there will be no other desktop environment after upgrade."),
```
Please apply the coding standards to the proposed change (max 120 chars one line).